pkg/fileutil/fs_case.go

```diff
@@ -0,0 +1,82 @@
+package fileutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	nanoid "github.com/matoous/go-nanoid/v2"
+)
+
+// caseSensitiveNamePrefix is a prefix used for testing case sensitivity.
+// It contains both lowercase and uppercase characters.
+const caseSensitiveNamePrefix = ".cAsE."
+
+// IsCaseInsensitiveLocation returns true if dirPath is a directory on a
+// case-insensitive filesystem.  dirPath must be writable.
+func IsCaseInsensitiveLocation(fs FS, dirPath string) (bool, error) {
+	// Random element in the filename to ensure uniqueness: the filename
+	// will not be in use from anything else.
+	id, err := nanoid.New()
+	if err != nil {
+		return false, fmt.Errorf("generate random name: %w", err)
+	}
+	path := filepath.Join(dirPath, caseSensitiveNamePrefix+id)
+	err = fs.Touch(path)
+	if err != nil {
+		return false, fmt.Errorf("touch %s: %w", path, err)
+	}
+	defer func() {
+		_ = fs.Remove(path)
```

Comment:
I'd like to find a solution to the problem where remove fails as it affects the locally synced directory. The file will be synced to remote and the user is not aware of it.
I understand we don't want to use tmpfile here because the tmp folder might be on a different FS. But I think we need a solution for this.

Reply:
This is literally not possible to fix.  OTOH it is also not possible to fail here unless a process with user privileges is actively trying to sabotage the user.  I could return an error and `panic`.

Reply:
Agreed f2f: print an warning message.